packages/block-editor/src/components/block-pattern-setup/index.js

```diff
@@ -22,43 +20,106 @@ import BlockPreview from '../block-preview';
 import SetupToolbar from './setup-toolbar';
 import usePatternsSetup from './use-patterns-setup';
 import { VIEWMODES } from './constants';
+import { unlock } from '../../lock-unlock';
+
+const {
+	CompositeV2: Composite,
+	CompositeItemV2: CompositeItem,
+	useCompositeStoreV2: useCompositeStore,
+} = unlock( componentsPrivateApis );
 
 const SetupContent = ( {
 	viewMode,
 	activeSlide,
 	patterns,
+	onChangeSlide,
 	onBlockPatternSelect,
 	showTitles,
 } ) => {
-	const composite = useCompositeState();
 	const containerClass = 'block-editor-block-pattern-setup__container';
-	if ( viewMode === VIEWMODES.carousel ) {
-		const slideClass = new Map( [
-			[ activeSlide, 'active-slide' ],
-			[ activeSlide - 1, 'previous-slide' ],
-			[ activeSlide + 1, 'next-slide' ],
-		] );
-		return (
-			<div className="block-editor-block-pattern-setup__carousel">
-				<div className={ containerClass }>
-					<ul className="carousel-container">
-						{ patterns.map( ( pattern, index ) => (
-							<BlockPatternSlide
-								className={ slideClass.get( index ) || '' }
-								key={ pattern.name }
-								pattern={ pattern }
-							/>
-						) ) }
-					</ul>
+	const commonProps = { containerClass, patterns };
+
+	return viewMode === VIEWMODES.carousel ? (
+		<SetupContentAsCarousel
+			{ ...commonProps }
+			activeSlide={ activeSlide }
+			onChangeSlide={ onChangeSlide }
+		/>
+	) : (
+		<SetupContentAsGrid
+			{ ...commonProps }
+			onBlockPatternSelect={ onBlockPatternSelect }
+			showTitles={ showTitles }
+		/>
+	);
```

Comment:
Refactoring these out a bit to make it a little clearer as to what is responsible for which behaviour (also makes `Composite` a bit happier!).